src/Authentication/Authentication/TokenCache/MacTokenCache.cs

```diff
@@ -0,0 +1,214 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph.PowerShell.Authentication.TokenCache
+{
+    using Microsoft.Graph.PowerShell.Authentication.TokenCache.NativePlatformLibs;
+    using System;
+    using System.Globalization;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// A set of methods for getting, setting and deleting tokens on MacOS via KeyChain.
+    /// </summary>
+    internal static class MacTokenCache
+    {
+        /// <summary>
+        /// Gets an app's token from MacOS KeyChain.
+        /// </summary>
+        /// <param name="appId">An app/client id.</param>
+        /// <returns>A decypted token.</returns>
+        public static byte[] GetToken(string appId)
+        {
+            IntPtr passwordDataPtr = IntPtr.Zero;
+            IntPtr itemPtr = IntPtr.Zero;
+
+            try
+            {
+                byte[] contentBuffer = new byte[0];
+                int resultStatus = MacNativeKeyChain.SecKeychainFindGenericPassword(
+                    keychainOrArray: IntPtr.Zero,
+                    serviceNameLength: (uint)Constants.TokenCahceServiceName.Length,
+                    serviceName: Constants.TokenCahceServiceName,
+                    accountNameLength: (uint)appId.Length,
+                    accountName: appId,
+                    passwordLength: out uint passwordLength,
+                    passwordData: out passwordDataPtr,
+                    itemRef: out itemPtr);
+
+                if (resultStatus == MacNativeKeyChain.SecResultCodes.errSecItemNotFound)
+                    return contentBuffer;
+
+                else if (resultStatus != MacNativeKeyChain.SecResultCodes.errSecSuccess)
```

Comment:
Opinion: I am not familiar with the current design guidelines of this repository, but it is hard to connect this `if` with the `else` block below without the curly brackets. To prevent accidental modifications, I personally prefer having the curly brackets even if there is a single statement in the block. If you have a reasoning against it, I would be happy to hear it.

Reply:
Even if you decide to keep it this way, you can remove `else` keyword, as the only statement above is a return.